src/App/App.js

```diff
@@ -4,10 +4,75 @@ import Favorites from '../Favorites/Favorites.js'
 import Form from '../Form/Form.js'
 import Home from '../Home/Home.js'
 import Nav from '../Nav/Nav.js'
+import Poster from '../Poster/Poster.js'
+import { mockQuotes, mockImages, mockPosters } from '../mockData'
+import {useState, useEffect } from 'react'
+import { getData } from '../apiCalls'
 
 const App = () => {
+  const [quotes, setQuotes] = useState(mockQuotes.quotes)
+  const [images, setImages] = useState(mockImages.images)
+
+  //// --> hopefully ready for real data - needs testing/debugging
+  // useEffect(() => {
+  //   getData(quotes)
+  //   .then(response => setQuotes(response.quotes))
+  //   .catch(error => alert(error.message))
+
+  //   getData(images)
+  //   .then(response => setImages(response.images))
+  //   .catch(error => alert(error.message))
+  // }, [])
+
+
+  const filterQuotes = (type) => {
+    return quotes.filter(quote => quote.type === type)
+  }
+  
+  const getRandom = (array) => {
+    const index = Math.floor(Math.random() * array.length)
+    return array[index]
+  }
+
+  const [poster, setPoster] = useState({});
+
+  const randomizePoster = (type) => {
+    const quote = getRandom(filterQuotes(type)).quote
+    const image = getRandom(images).src
+    
+    setPoster({
+      quote: quote,
+      image: image,
+      type: type
+    })
+  }
+
+  const [fontSize, setFontSize] = useState('');
+
+  const changeFontSize = (param) => {
+    setFontSize(param === 'poster' ? '2.5em' : '1em')
+  }
+
```

Comment:
I would love to have this syntax explained to me, because I can not wrap my head around it